# Dilute Weak Acid

The problem here is that solving dilute acid problems the same as concentrated strong acid problems can cause issues. For example, a strong acid at $10^{-8} F$ solved naively would give $pH = 8$, a basic value which is obviously wrong.

Let's take as our example HCl, with a $K_{a}=8000$ If you have $[HCl] = 10^{-6}$, then we should first find the amount of $[H_3O^+]$ FROM the acid - This is easily computed as $10^{-6}$

But now water autoprotolysis happens as well, so we should consider the reaction $2H_2O /leftrightharpoons H_3O^+ + /OH^-$. We should solve an ICE table here, with the initial concentration of $[H_3O^+] = 10^{-6}$. I'll just show the final result here:
$(10^{-6}+x)*x = K_w = 10^{-14}$
This solves to:

In [43]:
from sympy import *
x= symbols('x')
F=1e-6
solutions = solve(Eq((F+x)*x, 1e-14), x)
print('The solutions are:{}'.format(solutions))

The solutions are:[-1.00990195135928e-6, 9.90195135927848e-9]


We take the positive-valued answer of $9.902*10^{-9}$. This gives $[H_3O^+] = 1.0098*10^{-6}$

In [44]:
import numpy as np
C_H = F+9.902e-9
pH=-np.log10(C_H)
print("The pH is:{}".format(pH))

The pH is:5.99572076773


And $pH = 5.996$. This is slighly more acidic than using the acid concentration alone, because the autoprotolysis of water contributes non-negligibly in this case.  We can double-check that the $HCl K_a$ is still valid. Now that we know the pH, we can use alpha fractions to find how much $Cl^-$ and $HCl$ are still around:
$\alpha_{HCl} = \frac{[H^+]}{[H^+]+K_a}$ and $\alpha_{Cl^-} = \frac{K_a}{[H^+]+K_a}$
This gives an HCl alpha fractions and concentrations of:

In [45]:
K = 8000
F = 1e-6
alpha_HCl = C_H/(C_H + K)
print("The alpha for HCL is:{}".format(alpha_HCl))
alpha_Cl = K/(C_H + K)
print("The alpha for CL- is:{}".format(alpha_Cl))
C_HCl = alpha_HCl * F
print("The concentration for HCL is:{}M".format(C_HCl))
C_Cl = alpha_Cl * F
print("The concentration for CL- is:{}M".format(C_Cl))

The alpha for HCL is:1.26237749984e-10
The alpha for CL- is:0.999999999874
The concentration for HCL is:1.26237749984e-16M
The concentration for CL- is:9.99999999874e-07M


And together with $[H_3O^+]$ we can solve Q:

In [46]:
Q = C_H*C_Cl/C_HCl
print("Q = {}, which is equal to K".format(Q))

Q = 8000.0, which is equal to K


Now we will consider the case where $F_{HCl} = 10^{-8}$, using the same techniques.

In [47]:
F = 1e-8
solutions = solve(Eq((F+x)*x, 1e-14), x)
print('The solutions are:{}'.format(solutions))

The solutions are:[-1.05124921972504e-7, 9.51249219725039e-8]


Note that here, the amount of $H_3O^+$ from autoprotolysis is $9$ times greater than the amount from the acid. We'll do the same checks below:

In [48]:
C_H = F+9.512e-8
pH=-np.log10(C_H)
print("The pH is:{}".format(pH))
alpha_HCl = C_H/(C_H + K)
print("The alpha for HCL is:{}".format(alpha_HCl))
alpha_Cl = K/(C_H + K)
print("The alpha for CL- is:{}".format(alpha_Cl))
C_HCl = alpha_HCl * F
print("The concentration for HCL is:{}M".format(C_HCl))
C_Cl = alpha_Cl * F
print("The concentration for CL- is:{}M".format(C_Cl))
Q = C_H*C_Cl/C_HCl
print("Q = {}, which is equal to K".format(Q))

The pH is:6.97831464778
The alpha for HCL is:1.31399999998e-11
The alpha for CL- is:0.999999999987
The concentration for HCL is:1.31399999998e-19M
The concentration for CL- is:9.99999999987e-09M
Q = 8000.0, which is equal to K


Let's do it for a truly dilute acid, and a smaller $K_a$. Now that we have the code, we only need to copy and paste:

In [49]:
F = 1e-12
K = 10
solutions = solve(Eq((F+x)*x, 1e-14), x)
print('The solutions are:{}'.format(solutions))

The solutions are:[-1.00000500001250e-7, 9.99995000012500e-8]


In [50]:
C_H = F+9.99995e-8
pH=-np.log10(C_H)
print("The pH is:{}".format(pH))
alpha_HA = C_H/(C_H + K)
print("The alpha for HA is:{}".format(alpha_HA))
alpha_B = K/(C_H + K)
print("The alpha for CL- is:{}".format(alpha_B))
C_HA = alpha_HA * F
print("The concentration for HA is:{}M".format(C_HA))
C_B = alpha_B * F
print("The concentration for B- is:{}M".format(C_B))
Q = C_H*C_B/C_HA
print("Q = {}, which is equal to K".format(Q))

The pH is:6.99999782853
The alpha for HA is:1.00000499e-08
The alpha for CL- is:0.99999999
The concentration for HA is:1.00000499e-20M
The concentration for B- is:9.9999999e-13M
Q = 10.0, which is equal to K
